### PR TITLE
[LPT] Split & VariadicSplit support

### PR DESCRIPTION
**Pull request by v-Golubev**
Issue #48205

Validation:
• Accuracy: developer-services/job/accuracy/1507/ (reference: developer-services/job/accuracy/1506/)
• Performance: developer-services/job/performance/761/ (reference: developer-services/job/performance/745/)

Validation:
• Accuracy: developer-services/job/accuracy/1433/ (reference: developer-services/job/accuracy/1455/)
• Performance: developer-services/job/performance/730/ (reference: developer-services/job/performance/759/)

Validation:

• Accuracy: developer-services/job/accuracy/1147/ (reference: developer-services/job/accuracy/1160/)
• Accuracy: developer-services/job/accuracy/1162/ (reference: developer-services/job/accuracy/1160/)

### CODE REVIEW

File: inference-engine/src/low_precision_transformations/src/concat_multi_channels.cpp [hunk 1]
```diff
@@ -224,14 +228,13 @@ void ConcatMultiChannelsTransformation::fillQuantization(
                 // add concatenated dequantization operations to dequantization collection
                 dequantization.push_back(getConcatenatedDequantization(concat, dequantizationToConcatenate));
             } else {
-                std::shared_ptr<ngraph::opset1::StridedSlice> stridedSlice = ngraph::as_type_ptr<ngraph::opset1::StridedSlice>(parent);
-                if (stridedSlice) {
+                const size_t sourceOutputIdx = NetworkHelper::getParentOutputIndex(parent, layer);
+                if (parent->get_input_shape(0)[1] != parent->get_output_shape(sourceOutputIdx)[1]) {
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/concat_multi_channels.cpp:232]: 1. Check input shape, please: we can have one dimensional tensor (`Concat` parameter `axis="0"`). I think we need functional test case for this case too.
2. We should check `axis` parameter equality for all affected `Concat` operations. I think we can skip `Concat` operations with negative `axis` values. Can we do it in this PR?

File: inference-engine/src/low_precision_transformations/src/strided_slice.cpp [hunk 2]
```diff
@@ -45,13 +46,24 @@ std::shared_ptr<Node> stridedSliceDeqConstant(
...
+
+    auto beginMask = stridedSlice->get_begin_mask();
+    auto endMask = stridedSlice->get_end_mask();
+    for (size_t i = 0; i < constantShape.size(); ++i) {
+        // don't slice constant if current dimension is 1
+        if (constantShape[i] == 1ul) {
+            beginMask[i] = 1ul;
+            endMask[i] = 1ul;
+        }
+    }
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/strided_slice.cpp:58]: Cover by test case, please

File: inference-engine/src/low_precision_transformations/src/subgraph.cpp [hunk 3]
```diff
@@ -82,7 +50,7 @@ bool Subgraph::fillSubgraphForQuantization(
                 if (fakeQuantizeChild != nullptr) {
                     //
                 } else {
-                    if (layerTransformationsManager->isPrecisionPreserved(child) && isQuantizationPerChannel(child)) {
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/subgraph.cpp:85]: Check `Concat` `axes`, while `FakeQuantize` operation supports per channel & per tensor quantization only: for per channel quantization it has to be 1 and doesn't matter for per tensor quantization.

**Review by eshoguli — COMMENTED** [review]
Need offline meeting for `SplitTransformation` implementation discussion.

File: inference-engine/src/low_precision_transformations/src/strided_slice.cpp (hunk 2 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/strided_slice.cpp:58]: This test case has been added already (Please, see `inference-engine/tests/functional/inference_engine/lp_transformations/strided_slice_transformation.cpp`)

File: inference-engine/src/low_precision_transformations/src/concat_multi_channels.cpp (hunk 1 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/concat_multi_channels.cpp:232]: Thanks, fixed! Concats with unsupported `axis` values are not included in the subgraph now. Also I added normalization of negative axis values. So, negative `axis` values it is not problem

File: inference-engine/src/low_precision_transformations/src/subgraph.cpp (hunk 3 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/subgraph.cpp:85]: At the moment ConcatTransformation support only concat by channel (`axis = 1`). I think, we don't need to change this behaviour.

File: inference-engine/src/low_precision_transformations/src/subgraph.cpp [hunk 4]
```diff
@@ -221,6 +185,13 @@ bool Subgraph::empty() const {
 }
 
 bool Subgraph::fillSubgraphForConcat(const std::shared_ptr<ngraph::opset1::Concat>& concat, std::unordered_set<std::string>& handledLayers) {
+    const auto axis = concat->get_axis();
+    const size_t normalizedAxis = ngraph::normalize_axis(concat->get_friendly_name(), axis, concat->get_output_partial_shape(0).rank());
+    // supported only per-channel concat
+    if (normalizedAxis != 1ul) {
+        return false;
+    }
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/subgraph.cpp:193]: can we move the check from utility type `Subgraph` to `ConcatMultiChannelsTransformation` & `ConcatTransformation` transformations, please?

File: inference-engine/src/low_precision_transformations/src/split.cpp [hunk 5]
```diff
@@ -22,118 +23,69 @@ bool SplitTransformation::transform(TransformationContext& context, ngraph::patt
...
-    const auto mulSplitLengths = getConstSplitLengths(inputs, mulConstShape, outputSize);
+    int64_t axis = as_type_ptr<opset1::Constant>(split->get_input_node_shared_ptr(1))->cast_vector<int64_t>()[0];
+    size_t normalizeAxis = normalize_axis(split->get_friendly_name(), axis, split->get_input_partial_shape(0).rank());
 
-    std::vector<std::shared_ptr<ngraph::Node>> lastNodes(outputSize);
-    ngraph::OutputVector replacement;
-    for (size_t i = 0; i < outputSize; ++i) {
+    NodeVector lastNodes;
+    OutputVector replacement;
+    for (size_t i = 0; i < newSplit->get_output_size(); ++i) {
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/split.cpp:41]: I think to iterate for each output and fold `Split`/`VariadicSplit` for each output for the same dequantization operation repeatedly can be simpler for understanding, but I would like to decrease `fold` operations. Can you, please, refactor: fold `Split`/`VariadicSplit` only twice (`Subtract` and `Multiply` constants) and handle fold results for each output? Thanks!

**Review by eshoguli — COMMENTED** [review]
Thanks, it should work! But review comments and refactor, please: make source code simpler.

File: inference-engine/src/low_precision_transformations/src/split.cpp [hunk 6]
```diff
@@ -22,118 +23,69 @@ bool SplitTransformation::transform(TransformationContext& context, ngraph::patt
...
         Output<Node> previous = newSplit->output(i);
 
-        if (dequantization.convert != nullptr) {
-            const std::shared_ptr<ngraph::Node> convert =
-                dequantization.convert->clone_with_new_inputs({ newSplit->output(i) });
+        const auto splitConstant = [&](const std::shared_ptr<opset1::Constant> constant, const std::shared_ptr<Node> eltwise) {
+            if (constant->get_shape().empty()) {
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/split.cpp]: Done, thanks!

File: inference-engine/src/low_precision_transformations/src/network_helper.cpp [hunk 7]
```diff
@@ -86,6 +87,26 @@ bool NetworkHelper::isConstantPath(const std::shared_ptr<Node>& op) {
     return true;
 }
 
+std::shared_ptr<opset1::Constant> NetworkHelper::foldDequantizationConstant(
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/network_helper.cpp:90]: Discussed offline: method refactored and comments added. API has not been changed.

File: inference-engine/src/low_precision_transformations/src/split.cpp [hunk 8]
```diff
@@ -22,118 +23,69 @@ bool SplitTransformation::transform(TransformationContext& context, ngraph::patt
...
+                return constant;
+            } else {
+                // if batch is absent in constant shape - add batch
+                const auto normalizedConstant = NetworkHelper::normalizeDequantizationShape(eltwise);
+                if (normalizedConstant->get_shape()[normalizeAxis] == 1) {
+                    return normalizedConstant;
+                } else {
+                    return NetworkHelper::foldDequantizationConstant(newSplit, normalizedConstant, i);
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/split.cpp]: Thanks, done!

File: inference-engine/src/low_precision_transformations/src/split.cpp [hunk 9]
```diff
@@ -22,118 +23,69 @@ bool SplitTransformation::transform(TransformationContext& context, ngraph::patt
...
         }
-        const std::shared_ptr<ngraph::Node> multiply = std::make_shared<ngraph::opset1::Multiply>(previous, mulConst);
 
-        lastNodes.push_back(multiply);
-        replacement.push_back(multiply);
+        lastNodes.push_back(previous.get_node_shared_ptr());
+        replacement.push_back(previous);
     }
 
     replace_node(split, replacement);
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/split.cpp]: Discussed offline. The current implementation is correct and will not be changed

File: inference-engine/src/low_precision_transformations/src/subgraph.cpp (hunk 4 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/subgraph.cpp:224]: Discussed offline. It is impossible to move this check to `ConcatTransformation`. This check should call when subgraph filling.

File: inference-engine/src/low_precision_transformations/src/split.cpp (hunk 5 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/split.cpp]: Done, thanks!

**Review by eshoguli — APPROVED** [review]

**Pull request merged**